product_extended_segmentation/model/template.py

```diff
@@ -216,15 +215,19 @@ def _factor(factor, product_efficiency, product_rounding):
         if float_is_zero(diff, precision_id) or \
                 (current_price and diff < 0 and
                     computed_th > bottom_price_threshold):
-            tmpl_obj.message_post(cr, uid, [product_tmpl_id.id],
+            product_id = prod_obj.search(cr, uid, [('bom_ids', '=', bom.id)])
```

Comment:
By doing this search you could be excluding all those product.product records that are not linked to a BoM record